lib/model/workout.dart

```diff
@@ -9,21 +15,86 @@ class WorkoutExercise implements GenericModel {
   static const REST_BETWEEN_SETS = 'in-rest';
   static const AFTER_REST = 'post-rest';
 
+  BuildContext? context;
   int? exerciseId;
   int? equipmentUsed;
   int? setCount;
   int? time;
   int? restBetween;
   int? restAfter;
 
+  ExerciseBloc? exerciseBloc;
+
   WorkoutExercise({
+    this.context,
     this.exerciseId,
     this.equipmentUsed,
     this.setCount,
     this.time,
     this.restBetween,
     this.restAfter,
-  });
+  }) {
+    if (context != null) {
+      exerciseBloc = BlocProvider.read<ExerciseBloc>(context!);
+    }
+  }
+
+  Exercise get exercise {
+    if (exerciseId == null) {
+      throw ArgumentError(
+          'Exercise ID on Workout Exercise must not be null on runtime');
+    }
+    if (!exerciseBloc!.exerciseMap.containsKey(exerciseId)) {
+      throw ArgumentError('Exercise ID on Workout Exercise is not on Bloc');
+    }
+
+    return exerciseBloc!.exerciseMap[exerciseId]!;
+  }
+
+  // TODO: Make a better implementation on getting just one equipment name from bitwise value
+  // TODO: Validate equipment using repo?
+  String get equipmentUsedDisplayName {
+    if (equipmentUsed == null) {
+      throw ArgumentError(
+          'Equipment on Workout Exercise must not be null on runtime');
+    }
+    if (exercise.availableExercises == null) {
+      throw ArgumentError(
+          'Equipment of Exercise of Workout Exercise must not be null on runtime.');
+    }
+    if ((equipmentUsed! & exercise.availableExercises!) == 0) {
+      throw ArgumentError(
+          'Equipment of Workout Exercise is not available in Exercise.');
+    }
+
+    return convertIntToExerciseEquipment(equipmentUsed!)
+        .toList()[0]
+        .getDisplayText();
+  }
+
+  List<String> get muscleGroupDisplayNames {
+    return exercise.muscleGroupsSet
+        .map((muscleGroup) => muscleGroup.getDisplayText())
+        .toList();
+  }
+
+  int get totalTime {
+    if (time == null) {
+      throw ArgumentError(
+          'Time on Workout exercise must not be null on runtime');
+    } else if (setCount == null) {
```

Comment:
There's no need to chain if else statements since you already throw.

Might be better to just do assertions instead
``` dart
assert(time != null);
assert(setCount != null);
...
```